R/pkg/R/DataFrame.R

```diff
@@ -1266,6 +1266,83 @@ setMethod("dapplyCollect",
             ldf
           })
 
+#' gapply
+#'
+#' Group the SparkDataFrame using the specified columns and apply the R function to each
+#' group.
+#'
+#' @param x A SparkDataFrame
+#' @param cols Grouping columns
+#' @param func A function to be applied to each group partition specified by grouping
+#'             column of the SparkDataFrame. The function `func` takes as argument
+#'             a key - grouping columns and a data frame - a local R data.frame.
+#'             The output of `func` is a local R data.frame.
+#' @param schema The schema of the resulting SparkDataFrame after the function is applied.
```

Comment:
minor comment: It will be good to clarify how this schema can be constructed. i.e. something like `The output schema is usually the the schema for the key along with the schema of the output R data frame.` We can also highlight this in the programming guide